### DISCUSSION TITLE
How to format [custom_repo] for non-standard server URLs

### DISCUSSION THREAD

**Discussion by cbc02009** [nroi/flexo#121] [Q&A] [answered]
I use Flexo for my home Arch Linux installs already, but I would like to add caching for some extra custom repositories that don't seem to use the format that flexo expects for their URLs.

Examples:
```
Server = https://us.cachyos.org/repo/$arch/$repo
```
for this one the `$arch` and `$repo` are swapped from the formatting of the normal arch repos. Trying this in flexo returns a 404 from the cachyos server.

```
Server = https://us.cachyos.org/repo/$arch_v3/$repo
```
Cachyos also provides builds that are optimized for different processor versions (in this case zen3). So it adds extra characters after the `$arch` variable.

Is there any way to specify the full server URL in the flexo config?

**Comment by nroi** [comment]
> Is there any way to specify the full server URL in the flexo config?

I'm not sure if that makes any sense, because pacman generally sends `GET` requests that already include the path (which then includes the `$arch` and `$repo`). Therefore, flexo should just know the host name of the custom repo, because that is the only component of the URL that should be replaced.

> for this one the $arch and $repo are swapped from the formatting of the normal arch repos.

But why is that an issue? The `$arch` and `$repo` variables are specified in `/etc/pacman.conf`, you can just swap them there.

> Trying this in flexo returns a 404 from the cachyos server.
> Server = https://us.cachyos.org/repo/$arch_v3/$repo

Where did you specify this? If you specify this in `/etc/pacman.conf`, then pacman sends the request directly to cachyos.org, not to flexo.

As described in the [README](https://github.com/nroi/flexo?tab=readme-ov-file#using-unofficial-user-repositories), you have to adapt both your `/etc/pacman.conf` and your flexo config to use custom repositories. Please show both configs so I have a better understanding of the problem.

**Reply by cbc02009** [reply, marked as answer]
Hi @nroi ,

Thanks for the reply and sorry for the delayed response. Your message made me realize that the issue was a missing subpath in the pacman.conf URL, and not anything to do with Flexo. Should have been `Server = http://localhost:7878/custom_repo/cachyos/repo/$arch/$repo` and not `Server = http://localhost:7878/custom_repo/cachyos/$arch/$repo`. I was definitely overthinking it. Very much appreciate the help.
